### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by TahenyBELGUITH**
In this milestone  I do the following :

- Create a class Nameable
- Turn  Person class to Nameable
- Prepare base Decorator
- Prepare CapitalizeDecorator and TrimmerDecorator

a code reviewer is appreciated  😄 
@TahenyBELGUITH

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require_relative 'nameable'
+
+class BaseDecorator
+  def initialize(nameable)
+    @nameable = nameable
+  end
```
Comment (ShoiraTa) [inline base_decorator.rb:6]: - [ ] Please add `attr_accessor :nameable` to be able to access it 😉

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @TahenyBELGUITH ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

**First let me to point out:**

- Great commit history! ✔️ 
- You passed all the Linters! ✔️ 
- Professional pull request! ✔️ 

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username [@shoirata].


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: base_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require_relative 'nameable'
+
+class BaseDecorator
```
Comment (1GORDON) [inline base_decorator.rb:3]: Great job @TahenyBELGUITH 🤝

- [ ] [Optional] Just like the previous reviewer stated, kindly add the attribute accessor `attr_accessor :nameable`.

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @TahenyBELGUITH ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**